### DISCUSSION TITLE
watch uselazyfetch never gets updated

### DISCUSSION THREAD

**Discussion by ac615223s5** [nuxt/nuxt#16227] [Questions] [answered]
the doc says to put this ```const { pending, data: posts } = useLazyFetch('/api/posts') 

watch(posts, (newPosts) => {  // Because posts starts out null, you won't have access  // to its contents immediately, but you can watch it.})```

but the watch never gets called after the data arrives

**Comment by TheAlexLichter** [comment, marked as answer]
Update: [Working example](https://stackblitz.com/edit/nuxtframeworkdiscussions9464?file=app.vue)
---

Hey! Could you kindly provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Reply by ac615223s5** [reply]
`<template>`
  <div>
    <NuxtWelcome />
  </div>
</template>
<script setup>

const { data } = useLazyFetch('https://api.publicapis.org/entries');

console.log(1);

watch(data.value, () => {

  console.log('aaa');

});

</script>

**Reply by TheAlexLichter** [reply]
Please use a CodeSandbox or StackBlitz next time, as written in the linked post above ☺️

**Reply by TheAlexLichter** [reply]
[Works fine for me](https://stackblitz.com/edit/nuxtframeworkdiscussions9464?file=app.vue), but you should watch `data`, not `data.value`.

**Reply by ac615223s5** [reply]
Sorry i used it but didn't know how to link
I didn't see the console message aaa in your example?
![image](https://user-images.githubusercontent.com/33037196/205398564-77e4b5cd-2632-4738-be00-2b2ba3028efa.png)

**Reply by TheAlexLichter** [reply]
Whoops, sorry. Didn't save it. Added an `immediate: true` option and it should work fine there ☺️

**Reply by ac615223s5** [reply]
thanks
